src/utils.h

```diff
@@ -4,6 +4,9 @@
 
 #include <glib.h>
 
+#define STR_EMPTY(s) !STR_FULL(s)
+#define STR_FULL(s) (s && strnlen(s, 1))
```

Comment:
Isn't `strnlen(s,1)` the same as `s[0] != '\0'`?

Reply:
I mentioned it because it does the exact same thing and it saves a function call in the meantime. Its just a nitpick but good change to have IMO.